File: template/src/integrations/agent365Adapter.js
/**
 * Agent365 adapter placeholder.
 *
 * Replace TODOs with concrete imports and calls from the official Agent365 SDK
 * for your chosen language/runtime.
 */
export class Agent365Adapter {
  constructor(config) {
    this.config = config;

    // TODO: Initialize the Agent365 SDK client with tenant/app credentials.
    // Example placeholder:
    // this.client = new Agent365Client({
    //   tenantId: config.agent365TenantId,
    //   appId: config.agent365AppId,
    //   appSecret: config.agent365AppSecret,
    //   endpoint: config.agent365ReportingEndpoint
    // });
  }

  async reportTurnStart(context) {
    // TODO: Map to SDK event/logging API.
    return {
      turnId: context.turnId,
      status: "started"
    };
  }

  async reportPurviewDecision(context, stage, decision) {
    // TODO: Map to SDK event/logging API.
    return {
      turnId: context.turnId,
      stage,
      decision
    };
  }

  async reportTurnEnd(context, result) {
    // TODO: Map to SDK event/logging API.
    return {
      turnId: context.turnId,
      status: "completed",
      result
    };
  }
}

File: template/python/agent365_adapter.py
class Agent365Adapter:
    """
    Agent365 adapter placeholder.

    Replace TODOs with concrete imports and calls from the official Agent365 SDK.
    """

    def __init__(self, config: dict):
        self.config = config
        # TODO: Initialize your Agent365 SDK client here.

    async def report_turn_start(self, context: dict) -> dict:
        # TODO: Map to SDK event/logging API.
        return {"turnId": context["turn_id"], "status": "started"}

    async def report_purview_decision(self, context: dict, stage: str, decision: dict) -> dict:
        # TODO: Map to SDK event/logging API.
        return {"turnId": context["turn_id"], "stage": stage, "decision": decision}

    async def report_turn_end(self, context: dict, result: dict) -> dict:
        # TODO: Map to SDK event/logging API.
        return {"turnId": context["turn_id"], "status": "completed", "result": result}
